app/views/advanced_messenger/_global_notifications_render_functions.html.erb

```diff
@@ -0,0 +1,178 @@
+        const collapsed = readByUsers[currentUserId]?.collapsed;
+
+        if (!isOfInterestForCurrentUser && !collapsed || isReadByCurrentUser == 1) {
+            renderExpandedCollapsed(journalId, true);
+        } else {
+            renderExpandedCollapsed(journalId, false, isInitialRender);
+        }
+        
+        // ====== Current user read status + read/unread button ======
+        if (isOfInterestForCurrentUser) {
+            let span = currentNote.find('.read-unread-indicators-at-right').find('.read_checkbox_placeholder');
+            span.html("<span style='padding-right: 5px'> <%=t(:journal_read)%> </span>"
+                        + (isReadByCurrentUser ? 
+                        "<a class='icon icon-checked-green readButton' data-remote='true' rel='nofollow' data-method='post' href='/advanced_messenger/" + journalId + "/0/update_read_by_users'></a>" 
+                      : "<a class='icon icon-unchecked-red unreadButton' data-remote='true' rel='nofollow' data-method='post' href='/advanced_messenger/" + journalId + "/1/update_read_by_users'></a>"));
+        }
+
+        // ====== Panel border color ======
+        if (!isOfInterestForCurrentUser) {
+            currentNote.css({"border-color": "#628DB6"});
+        } else if (isReadByCurrentUser == 0) {
+             currentNote.css({"border-color": RED});
+        } else {
+            currentNote.css({"border-color": GREEN});
+        }
+
+        // ====== note-preview ======
+        renderNotesPreviewContent(journalId, isOfInterestForCurrentUser, isReadByCurrentUser, currentUserId, notesPreview);
+
+    }
+
+    function renderNotesPreviewContent(journalId, isOfInterestForCurrentUser, isReadByCurrentUser, currentUserId, notesPreview) {
+        var currentNote = $("#change-" + journalId + ".journal.has-notes");
+        const notesPreviewDiv = currentNote.find('.note-preview');
+        if (isOfInterestForCurrentUser) {
+            if (isReadByCurrentUser) {
+                notesPreviewDiv.html("<span class='collapse-note-read-message'><%=t(:journal_collapse_note_message_read)%></span><span class='collapse-note-preview'>" + notesPreview + "...</span>");
+            } else {
+                notesPreviewDiv.html("<span class='collapse-note-unread-message'><%=t(:journal_collapse_note_message_unread)%></span>");
+            }
+        } else {
+            notesPreviewDiv.html("<span class='collapse-note-message'><%=t(:journal_collapse_note_message)%></span><span class='collapse-note-preview'>" + notesPreview + "...</span>");
+        }
+    }
+
+    function renderReadIndicatorsForAllUsers(journalId, currentUserId, journalCreationDate, users, readStatuses) {
+        var currentNote = $("#change-" + journalId + ".journal.has-notes");
+        let userInitials;
+        let color;
+        let tooltipUserList = "";
+        let usersColoredInitials = "<span style='padding-right: 5px'>";
+        let inSomeoneInterestedIn = Object.keys(users).length > 0;
```

Comment:
I think you wanted to say `isSomeoneInterestedIn`, right?

Reply:
Yes.